### PR TITLE
Interactions.py : Add deprecation message for section parameter

### PR DESCRIPTION
**Pull request by alphadose**
The documentation for section parameter was not provided
in the docstring. This commit adds the documentation.

Closes https://github.com/coala/coala/issues/4740

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: coalib/output/Interactions.py [hunk 1]
```diff
@@ -10,6 +10,7 @@ def fail_acquire_settings(log_printer, settings_names_dict, section):
                                 a list containing a description in [0] and the
                                 name of the bears who need this setting in [1]
                                 and following.
+    :param section:             The section the action corresponds to.
```
Comment (Makman2) [inline coalib/output/Interactions.py]: What "action"? :)

Comment (alphadose) [reply coalib/output/Interactions.py]: It is deprecated so can I state that in the docstring . I cannot get a hold of its functionality and according to previous commits this was supposed to be its documentation.

**Comment by Makman2** [comment]
We should also add a log message inside code:
```
if section is not None:
    logging.warn('fail_acquire_settings: section parameter is deprecated.')
```
or a similar message.

**Comment by alphadose** [comment]
@Makman2 I think we need to modify the tests/output/InteractionsTest.py on account of new warning log message placed due to section parameter deprecation . If I am wrong then I would be grateful if you point out why the builds are not passing .

**Comment by alphadose** [comment]
@Makman2 The builds are passing on account of the change introduced in tests/output/InteractionsTest.py . Is this okay ?

File: tests/output/InteractionsTest.py [hunk 2]
```diff
@@ -4,14 +4,13 @@
...
 from coalib.output.printers.LogPrinter import LogPrinter
-from coalib.settings.Section import Section
 
 
 class InteractionsTest(unittest.TestCase):
 
     def test_(self):
         log_printer = LogPrinter(NullPrinter())
-        section = Section('')
+        section = None
```
Comment (Makman2) [inline tests/output/InteractionsTest.py]: I would add another test function `test_section_deprecation` which catches the warning message `fail_acquire_settings: section parameter is deprecated.`
You can use
```
from testfixtures import LogCapture

with LogCapture() as capture:
    capture.check(
            ('root', 'INFO', 'Sample message 1'),
            ('root', 'INFO', 'Sample message 2'),
            ('root', 'INFO', 'Sample message 3')
    )
```

**Comment by Makman2** [comment]
please squash the commits together ;)

**Comment by alphadose** [comment]
@Makman2 can you please elaborate on how to use the log capture? I tried using it in another test function and also after each assert statement but the builds are still failing.

**Comment by Makman2** [comment]
So in fact you write the code under test into the with-statement with the LogCapture object, and then you use `capture.check`:
```python
with LogCapture() as capture:
    # Call here fail_acquire_settings that trigger the warning message
    capture.check(
        ('root', 'WARN', '<the message>')
    )
```
You can always print out what was captured using `print` (the `capture` object should have some kind of field for that, like `print(capture.messages)` or so) so you can witness with your own eyes what was actually logged ;)

**Comment by alphadose** [comment]
@Makman2 please review :)

File: tests/output/InteractionsTest.py [hunk 3]
```diff
@@ -12,11 +13,33 @@ class InteractionsTest(unittest.TestCase):
...
+               'WARNING',
+               'fail_acquire_settings: section parameter'
+               ' is deprecated.'),
+              ('root',
+               'ERROR',
+               'During execution, we found that some required settings were'
+               ' not provided. '
+               'They are:\n'
+               'setting (from bear) - description')
+            )
```
Comment (Makman2) [inline tests/output/InteractionsTest.py]: I think you should keep the test case as is, except removing `section` from the function calls. Then you introduce a new test function checking the deprecation message.

**Comment by Makman2** [comment]
Could you improve your commit message and remove the duplicated sentences? :)

**Comment by nvzard** [comment]
I think there is a typo in the commit message( `Interactions..`)
> Add a log capture functionality to InteractionsTest.py
to capture the error and warning logs as a result of
testing Interactions..

File: tests/output/InteractionsTest.py (hunk 3 above)
Comment (Makman2) [reply tests/output/InteractionsTest.py]: bump ;)

File: tests/output/InteractionsTest.py [hunk 4]
```diff
@@ -12,11 +13,35 @@ class InteractionsTest(unittest.TestCase):
...
```
[the remark this answers is not in the document]
Comment (newbazz) [reply tests/output/InteractionsTest.py]: Add a new line at the end of file.

[the remark this answers is not in the document]
Comment (newbazz) [reply tests/output/InteractionsTest.py]: Run PEP8 bearn.

**Comment by alphadose** [comment]
@Makman2 please review  :) . PS:- Sorry for the late response

File: tests/output/InteractionsTest.py [hunk 5]
```diff
@@ -12,11 +13,35 @@ class InteractionsTest(unittest.TestCase):
     def test_(self):
         log_printer = LogPrinter(NullPrinter())
         section = Section('')
-        self.assertRaises(TypeError, fail_acquire_settings, log_printer, None,
-                          section)
+        self.assertRaises(TypeError, fail_acquire_settings, log_printer,
+                          None, section)
```
Comment (Makman2) [inline tests/output/InteractionsTest.py]: unnecessary change

File: tests/output/InteractionsTest.py [hunk 6]
```diff
@@ -12,11 +13,35 @@ class InteractionsTest(unittest.TestCase):
...
+
```
Comment (Makman2) [inline tests/output/InteractionsTest.py]: also unnecessary change

File: tests/output/InteractionsTest.py [hunk 7]
```diff
@@ -12,11 +13,35 @@ class InteractionsTest(unittest.TestCase):
     def test_(self):
         log_printer = LogPrinter(NullPrinter())
         section = Section('')
```
Comment (Makman2) [inline tests/output/InteractionsTest.py]: I would just remove this variable and wouldn't call `fail_acquire_settings` with `section`, so just
```
self.assertRaises(TypeError, fail_acquire_settings, log_printer, None)
```
Same below.

PS: Ignore the "unnecessary change" comments then, as you have to modify these lines then :+1:

Comment (Makman2) [reply tests/output/InteractionsTest.py]: Ah and `fail_acquire_settings(..., ..., None)` just becomes `fail_acquire_settings(..., ...)` (see occurrence below)

File: tests/output/InteractionsTest.py [hunk 8]
```diff
@@ -12,11 +13,35 @@ class InteractionsTest(unittest.TestCase):
...
+        self.assertEqual(fail_acquire_settings(log_printer, {}, None),
+                         None, section)
+
+    def test_section_deprecation(self):
+
+        with LogCapture() as capture:
+
+            self.test_()
+            capture.check(
+              ('root',
```
Comment (Makman2) [inline tests/output/InteractionsTest.py]: indent: 4 spaces relative to opening statement:
```python
capture.check(
    ('...'    # <-- note: 4 leading spaces relative to `capture`
```

File: tests/output/InteractionsTest.py [hunk 9]
```diff
@@ -12,11 +13,35 @@ class InteractionsTest(unittest.TestCase):
...
+               'WARNING',
+               'fail_acquire_settings: section parameter'
+               ' is deprecated.'),
+              ('root',
+               'ERROR',
+               'During execution, we found that some required settings were'
+               ' not provided. '
+               'They are:\n'
+               'setting (from bear) - description')
+                          )
```
Comment (Makman2) [inline tests/output/InteractionsTest.py]: brace should close on same indent level as opening statement (so aligned to beginning of `capture...`)

File: tests/output/InteractionsTest.py [hunk 10]
```diff
@@ -12,11 +13,35 @@ class InteractionsTest(unittest.TestCase):
...
+            self.test_()
+            capture.check(
+              ('root',
+               'WARNING',
+               'fail_acquire_settings: section parameter'
+               ' is deprecated.'),
+              ('root',
+               'WARNING',
+               'fail_acquire_settings: section parameter'
+               ' is deprecated.'),
```
Comment (Makman2) [inline tests/output/InteractionsTest.py]: optional: I usually prefer putting the space into the string from the previous line, but as you like :) just a note :+1:

File: tests/output/InteractionsTest.py [hunk 11]
```diff
@@ -4,19 +4,36 @@
...
 
 class InteractionsTest(unittest.TestCase):
 
     def test_(self):
         log_printer = LogPrinter(NullPrinter())
-        section = Section('')
-        self.assertRaises(TypeError, fail_acquire_settings, log_printer, None,
-                          section)
+        self.assertRaises(TypeError, fail_acquire_settings, log_printer,
+                          None)
```
Comment (Makman2) [inline tests/output/InteractionsTest.py]: I think you can put this into the previous line too :)

File: tests/output/InteractionsTest.py [hunk 12]
```diff
@@ -4,19 +4,36 @@
...
+
```
Comment (Makman2) [inline tests/output/InteractionsTest.py]: not necessary to add this line :)

File: tests/output/InteractionsTest.py [hunk 13]
```diff
@@ -4,19 +4,36 @@
...
         self.assertRaises(AssertionError,
                           fail_acquire_settings,
                           log_printer,
-                          {'setting': ['description', 'bear']}, section)
-        self.assertEqual(fail_acquire_settings(log_printer, {}, section), None,
-                         section)
+                          {'setting': ['description', 'bear']})
+
```
Comment (Makman2) [inline tests/output/InteractionsTest.py]: if possible also remove this empty line :+1:

File: tests/output/InteractionsTest.py [hunk 14]
```diff
@@ -4,19 +4,36 @@
...
+        self.assertEqual(fail_acquire_settings(log_printer, {}, None),
```
Comment (Makman2) [inline tests/output/InteractionsTest.py]: omit the `None` inside the call of `fail_acquire_settings`

File: tests/output/InteractionsTest.py [hunk 15]
```diff
@@ -4,19 +4,36 @@
...
+                         None)
+
+    def test_section_deprecation(self):
+
+        section = Section('')
+        log_printer = LogPrinter(NullPrinter())
+
+        with LogCapture() as capture:
+
```
Comment (Makman2) [inline tests/output/InteractionsTest.py]: unnecessary newline

File: tests/output/InteractionsTest.py [hunk 16]
```diff
@@ -4,19 +4,36 @@
...
-                          {'setting': ['description', 'bear']}, section)
-        self.assertEqual(fail_acquire_settings(log_printer, {}, section), None,
-                         section)
+                          {'setting': ['description', 'bear']})
+
+        self.assertEqual(fail_acquire_settings(log_printer, {}, None),
+                         None)
+
+    def test_section_deprecation(self):
+
```
Comment (Makman2) [inline tests/output/InteractionsTest.py]: unnecessary newline

File: tests/output/InteractionsTest.py [hunk 17]
```diff
@@ -4,19 +4,36 @@
...
+        log_printer = LogPrinter(NullPrinter())
+
+        with LogCapture() as capture:
+
+            fail_acquire_settings(log_printer, {}, section)
+            capture.check(
+                ('root',
+                 'WARNING',
+                 'fail_acquire_settings: section parameter '
+                 'is deprecated.'),
```
Comment (Makman2) [inline tests/output/InteractionsTest.py]: optional: trailing comma not required, but you can keep it if you like :)

File: tests/output/InteractionsTest.py [hunk 18]
```diff
@@ -4,19 +4,36 @@
...
-                          section)
+        self.assertRaises(TypeError, fail_acquire_settings, log_printer,
+                          None)
+
         self.assertRaises(AssertionError,
                           fail_acquire_settings,
                           log_printer,
-                          {'setting': ['description', 'bear']}, section)
-        self.assertEqual(fail_acquire_settings(log_printer, {}, section), None,
-                         section)
```
Comment (Makman2) [inline tests/output/InteractionsTest.py]: oh interesting, actually this was coded badly before, the `section` parameter here is for the `message` parameter of `assertEqual`. If you like, you can remove it with a separate commit to make your changes more atomic (would be quite good I think + you get one commit credited more :D)

File: tests/output/InteractionsTest.py [hunk 19]
```diff
@@ -4,19 +4,29 @@
...
-        section = Section('')
-        self.assertRaises(TypeError, fail_acquire_settings, log_printer, None,
-                          section)
+        self.assertRaises(TypeError, fail_acquire_settings, log_printer, None)
         self.assertRaises(AssertionError,
                           fail_acquire_settings,
                           log_printer,
-                          {'setting': ['description', 'bear']}, section)
-        self.assertEqual(fail_acquire_settings(log_printer, {}, section), None,
-                         section)
```
Comment (PlentyMore) [inline tests/output/InteractionsTest.py]: Are you sure not to remove the `section` parameter with a separate commit ? Up to you :D)

**Comment by alphadose** [comment]
@Makman2 please review :)

**Comment by PlentyMore** [comment]
The PR title is not the same as the second commit message, the rest LGTM

**Comment by alphadose** [comment]
@newbazz  @Makman2 has said that it is better to use 2 commits for easier reviewing . One for removing the section param in the test file and one for adding the deprecation function .

**Comment by alphadose** [comment]
@Makman2

**Comment by newbazz** [comment]
@alphadose ohh is it.

File: tests/output/InteractionsTest.py [hunk 20]
```diff
@@ -1,9 +1,10 @@
 import unittest
 
 from pyprint.NullPrinter import NullPrinter
-
+from testfixtures import LogCapture
```
Comment (Makman2) [inline tests/output/InteractionsTest.py]: add a newline here (so we separate site-packages imports from coala imports)

**Comment by Makman2** [comment]
- About two separate commits: I think I meant something different, however I can't really remember any more (the commits were a bit different before). In this case I would now squash them together, so everything deprecation related happens in one go. Hopefully I don't confuse you^^ @newbazz you were right, squashing is good in this case
- https://github.com/coala/coala/pull/4744/commits/9cccd4befcd149045db4bc429d5d35985e042b18#r148324474
- Why did you remove the param description for `section`? I think it was quite good mentioning the deprecation :+1:

**Comment by Makman2** [comment]
`InteractionsTest.py: Remove section parameter` commit message body:
- remove the `Fixes ....` reference, as this commit is not fixing the issue
- > message in self.assertEqual is no longer required.

  It was never necessary, it was probably added by mistake. It's just not informative and will be displayed anyway when the assertion fails :) Please update the description accordingly :+1:

**Comment by Makman2** [comment]
`Interactions.py: Add msg for section deprecation`
- I would change to `Interactions.py: Deprecate section parameter`
  and remove the commit body altogether (except the issue reference). I think that would be precise enough :)

**Comment by Makman2** [comment]
ack f37061d 8ff5e34

**Comment by Makman2** [comment]
@gitmate-bot rebase

**Comment by Makman2** [comment]
@rultor merge
